ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter-input/filter-date-inputs/filter-relative-time-input/relativeTimeHelper.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+export const serialize = ({ last, unit }) => {
+  if (unit === undefined || !parseFloat(last)) {
+    return
+  }
+  const prefix = unit === 'm' || unit === 'h' ? 'PT' : 'P'
+  return `RELATIVE(${prefix + last + unit.toUpperCase()})`
+}
+
+export const deserialize = value => {
+  if (typeof value !== 'string') {
+    return
+  }
+
+  const match = value.match(/RELATIVE\(Z?([A-Z]*)(\d+\.*\d*)(.)\)/)
+  if (!match) {
+    return
+  }
+
+  let [, prefix, last, unit] = match
```

Comment:
is this valid syntax? if so, what does it do?

Reply:
I copied this logic from this file and didn't really mess with it too much: https://github.com/codice/ddf/blob/b7f6b8a34607eeb7eda6ff467c41814335cee8d0/ui/packages/catalog-ui-search/src/main/webapp/component/relative-time/serial.js#L16

But this syntax is array destructuring where the first element is ignored.

For example this line:
`let [, x, y] = [1, 2, 3]`
would ignore, 1, set x to 2 and y to 3